Run coretests and alloctests with cg_clif in CI

bjorn3 opened the pull request:
Part of https://github.com/rust-lang/rustc_codegen_cranelift/issues/1290

bjorn3 committed:
Unset RUSTC_WRAPPER in cg_clif's build system

bjorn3 committed:
Run coretests and alloctests with cg_clif in CI

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14197533394/job/39776243393) [(plain)](https://github.com/rust-lang/rust/commit/7845fc566f540c65115188e989154fdea5f099ec/checks/39776243393/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#19 exporting to docker image format
#19 sending tarball 20.8s done
#19 DONE 25.2s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'build.print-step-timings', '--enable-verbose-tests', '--set', 'build.metrics', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: build.build          := x86_64-unknown-linux-gnu
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
  Number of decisions:   4447
  longest path:          1159 (code:    152)
  longest backtrack:       66 (code:    428)
Shared 86733 out of 152951 states by creating 14756 new states, saving 71977
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc: In function ‘rtx_def* extract_bit_field_1(rtx, poly_uint64, poly_uint64, int, rtx, machine_mode, machine_mode, bool, bool, rtx_def**)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc:1864:45: warning: ‘*(unsigned int*)((char*)&imode + offsetof(scalar_int_mode, scalar_int_mode::m_mode))’ may be used uninitialized [-Wmaybe-uninitialized]
 1864 |       rtx sub = extract_bit_field_as_subreg (mode1, op0, imode,
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
 1865 |                                              bitsize, bitnum);
      |                                              ~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc:1824:19: note: ‘*(unsigned int*)((char*)&imode + offsetof(scalar_int_mode, scalar_int_mode::m_mode))’ was declared here
 1824 |   scalar_int_mode imode;
      |                   ^~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gimple-range-gori.cc: In member function ‘void range_def_chain::dump(FILE*, basic_block, const char*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gimple-range-gori.cc:319:19: warning: format not a string literal and no format arguments [-Wformat-security]
  319 |           fprintf (f, prefix);
      |           ~~~~~~~~^~~~~~~~~~~
---
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc:7930:9: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 7930 |   write (fd, "\n\n", 2);
      |   ~~~~~~^~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc: In member function ‘void driver::final_actions() const’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc:9307:13: warning: ignoring return value of ‘int truncate(const char*, __off_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 9307 |     truncate(totruncate_file, 0);
      |     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto-wrapper.cc: In function ‘bool find_and_merge_options(int, off_t, const char*, vec<cl_decoded_option>, bool, vec<cl_decoded_option>*, const char*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto-wrapper.cc:1165:8: warning: ignoring return value of ‘ssize_t read(int, void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1165 |   read (fd, data, length);
      |   ~~~~~^~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc: In function ‘void lto_resolution_read(splay_tree, FILE*, lto_file*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2091:10: warning: ignoring return value of ‘int fscanf(FILE*, const char*, ...)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2091 |   fscanf (resolution, " ");   /* Read white space.  */
      |   ~~~~~~~^~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2093:9: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2093 |   fread (obj_name, sizeof (char), name_len, resolution);
      |   ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2113:10: warning: ignoring return value of ‘int fscanf(FILE*, const char*, ...)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2113 |   fscanf (resolution, "%u", &num_symbols);
      |   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/jit/jit-recording.cc:32:
---
Applying io_quotes_use            to linux/blkzoned.h
Applying io_quotes_use            to linux/ipmi.h
Applying io_quotes_use            to linux/psp-dbc.h
Applying io_quotes_use            to linux/bt-bmc.h
Applying io_quotes_use            to linux/tps6594_pfsm.h
Applying io_quotes_use            to linux/cxl_mem.h
Applying io_quotes_use            to linux/wmi.h
Applying io_quotes_use            to linux/auto_fs.h
Applying io_quotes_use            to linux/mmtimer.h
Applying io_quotes_use            to linux/f2fs.h
Applying io_quotes_use            to linux/vhost.h
---
Applying machine_name             to x86_64-linux-gnu/bits/unistd_ext.h
Applying io_quotes_use            to x86_64-linux-gnu/asm/mtrr.h
Applying io_quotes_use            to x86_64-linux-gnu/asm/amd_hsmp.h
Applying machine_name             to openssl/e_os2.h
Applying io_quotes_use            to drm/xe_drm.h
Applying io_quotes_use            to drm/radeon_drm.h
Applying io_quotes_use            to drm/panfrost_drm.h
Applying io_quotes_use            to drm/etnaviv_drm.h
Applying io_quotes_use            to drm/lima_drm.h
Applying io_quotes_use            to drm/qaic_accel.h
Applying io_quotes_use            to drm/vc4_drm.h
Applying io_quotes_use            to drm/i915_drm.h
Applying io_quotes_use            to drm/omap_drm.h
Applying io_quotes_use            to drm/pvr_drm.h
Applying io_quotes_use            to drm/amdgpu_drm.h
Applying io_quotes_use            to drm/vgem_drm.h
Applying io_quotes_use            to drm/msm_drm.h
Applying io_quotes_use            to drm/v3d_drm.h
Applying io_quotes_use            to drm/exynos_drm.h
Applying io_quotes_use            to drm/nouveau_drm.h
Applying io_quotes_use            to drm/drm.h
Applying io_quotes_use            to drm/habanalabs_accel.h
Applying io_quotes_use            to drm/tegra_drm.h
Applying io_quotes_use            to rdma/rdma_user_ioctl.h
cc1: note: self-tests are not enabled in this build
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/c++tools/server.cc: In function ‘void server(bool, int, module_resolver*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/c++tools/server.cc:620:10: warning: ignoring return value of ‘int pipe(int*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
---
##[endgroup]
##[group][AOT] raw-dylib
[AOT] raw-dylib
##[endgroup]
##[group][TEST] sysroot
[TEST] sysroot
[COPY] sysroot_tests source
[PATCH] "sysroot_tests" <- "0027-sysroot_tests-128bit-atomic-operations.patch"
[PATCH] "sysroot_tests" <- "0028-sysroot_tests-Disable-long-running-tests.patch"
---
[RUSTC-TIMING] vec_deque_alloc_error test:true 0.303
error: unexpected `cfg` condition value: `panic_immediate_abort`
##[error]  --> alloctests/../alloc/src/raw_vec/mod.rs:26:16
   |
26 | #[cfg_attr(not(feature = "panic_immediate_abort"), inline(never))]
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove the condition
   |
   = note: no expected values for `feature`
   = help: consider adding `panic_immediate_abort` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
---

error: unexpected `cfg` condition value: `blah`
##[error]  --> coretests/tests/macros.rs:13:9
   |
13 |         feature = "blah" => {
   |         ^^^^^^^---------
   |         |
   |         help: remove the condition
   |
   = note: no expected values for `feature`
   = help: consider adding `blah` as a feature in `Cargo.toml`
   = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

error: unexpected `cfg` condition value: `foo`
##[error]   --> coretests/tests/macros.rs:125:9
    |
125 |         feature = "foo" => { fn works2() -> bool { false } }
    |         ^^^^^^^--------
    |         |
    |         help: remove the condition
    |
    = note: no expected values for `feature`
    = help: consider adding `foo` as a feature in `Cargo.toml`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

error: unexpected `cfg` condition value: `foo`
##[error]   --> coretests/tests/macros.rs:131:9
    |
131 |         feature = "foo" => { fn works3() -> bool { false } }
    |         ^^^^^^^--------
    |         |
    |         help: remove the condition
    |
    = note: no expected values for `feature`
    = help: consider adding `foo` as a feature in `Cargo.toml`
    = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration

error: unexpected `cfg` condition value: `foo`
##[error]   --> coretests/tests/macros.rs:143:9
    |
143 |         feature = "foo" => { fn works5() -> bool { false } }
    |         ^^^^^^^--------
    |         |
    |         help: remove the condition
    |
    = note: no expected values for `feature`
---

error: associated function `new` is never used
##[error]  --> alloctests/tests/sort/ffi_types.rs:11:12
   |
10 | impl FFIOneKibiByte {
   | ------------------- associated function in this implementation
11 |     pub fn new(val: i32) -> Self {
   |            ^^^

error: associated function `new` is never used
---

error: function `sort` is never used
##[error]  --> alloctests/tests/sort/known_good_stable_sort.rs:23:8
   |
23 | pub fn sort<T: Ord>(v: &mut [T]) {
   |        ^^^^

error: function `stable_sort` is never used
##[error]  --> alloctests/tests/sort/known_good_stable_sort.rs:28:4
   |
28 | fn stable_sort<T, F: FnMut(&T, &T) -> bool>(v: &mut [T], mut is_less: F) {
   |    ^^^^^^^^^^^

error: function `mergesort_main` is never used
##[error]  --> alloctests/tests/sort/known_good_stable_sort.rs:51:11
   |
51 | unsafe fn mergesort_main<T, F: FnMut(&T, &T) -> bool>(v: &mut [T], is_less: &mut F) {
   |           ^^^^^^^^^^^^^^

error: function `mergesort_core` is never used
##[error]  --> alloctests/tests/sort/known_good_stable_sort.rs:69:11
   |
69 | unsafe fn mergesort_core<T, F: FnMut(&T, &T) -> bool>(
   |           ^^^^^^^^^^^^^^

error: function `merge` is never used
##[error]   --> alloctests/tests/sort/known_good_stable_sort.rs:100:11
    |
100 | unsafe fn merge<T, F>(v: &mut [T], scratch_ptr: *mut T, is_less: &mut F, mid: usize)
    |           ^^^^^

error: function `unwrap_unchecked` is never used
##[error]   --> alloctests/tests/sort/known_good_stable_sort.rs:146:11
    |
146 | unsafe fn unwrap_unchecked<T>(opt_val: Option<T>) -> T {
    |           ^^^^^^^^^^^^^^^^

error: struct `BufGuard` is never constructed
##[error]   --> alloctests/tests/sort/known_good_stable_sort.rs:161:8
    |
161 | struct BufGuard<T> {
    |        ^^^^^^^^

error: associated function `new` is never used
##[error]   --> alloctests/tests/sort/known_good_stable_sort.rs:168:15
    |
166 | impl<T> BufGuard<T> {
    | ------------------- associated function in this implementation
167 |     // SAFETY: The caller has to ensure that len is not 0 and that T is not a ZST.
168 |     unsafe fn new(len: usize) -> Self {
    |               ^^^

error: function `random` is never used
##[error]  --> alloctests/tests/sort/patterns.rs:16:8
   |
16 | pub fn random(len: usize) -> Vec<i32> {
   |        ^^^^^^

error: function `random_uniform` is never used
##[error]  --> alloctests/tests/sort/patterns.rs:24:8
   |
24 | pub fn random_uniform<R>(len: usize, range: R) -> Vec<i32>
   |        ^^^^^^^^^^^^^^

error: function `random_zipf` is never used
##[error]  --> alloctests/tests/sort/patterns.rs:37:8
   |
37 | pub fn random_zipf(len: usize, exponent: f64) -> Vec<i32> {
   |        ^^^^^^^^^^^

error: function `random_sorted` is never used
##[error]  --> alloctests/tests/sort/patterns.rs:47:8
   |
47 | pub fn random_sorted(len: usize, sorted_percent: f64) -> Vec<i32> {
   |        ^^^^^^^^^^^^^

error: function `all_equal` is never used
##[error]  --> alloctests/tests/sort/patterns.rs:67:8
   |
---

error: function `saw_mixed` is never used
##[error]  --> alloctests/tests/sort/patterns.rs:90:8
   |
90 | pub fn saw_mixed(len: usize, saw_count: usize) -> Vec<i32> {
   |        ^^^^^^^^^

error: function `saw_mixed_range` is never used
##[error]   --> alloctests/tests/sort/patterns.rs:115:8
    |
115 | pub fn saw_mixed_range(len: usize, range: std::ops::Range<usize>) -> Vec<i32> {
    |        ^^^^^^^^^^^^^^^

error: function `pipe_organ` is never used
##[error]   --> alloctests/tests/sort/patterns.rs:154:8
    |
154 | pub fn pipe_organ(len: usize) -> Vec<i32> {
    |        ^^^^^^^^^^

error: function `get_or_init_rand_seed` is never used
##[error]   --> alloctests/tests/sort/patterns.rs:169:8
    |
169 | pub fn get_or_init_rand_seed() -> u64 {
    |        ^^^^^^^^^^^^^^^^^^^^^

error: static `SEED_VALUE` is never used
##[error]   --> alloctests/tests/sort/patterns.rs:180:8
    |
180 | static SEED_VALUE: OnceLock<u64> = OnceLock::new();
    |        ^^^^^^^^^^

error: function `rand_root_seed` is never used
##[error]   --> alloctests/tests/sort/patterns.rs:183:4
    |
183 | fn rand_root_seed() -> u64 {
    |    ^^^^^^^^^^^^^^

error: function `random_vec` is never used
##[error]   --> alloctests/tests/sort/patterns.rs:208:4
    |
208 | fn random_vec(len: usize) -> Vec<i32> {
    |    ^^^^^^^^^^

error: associated function `new` is never used
##[error]  --> alloctests/tests/sort/zipf.rs:28:12
   |
23 | impl ZipfDistribution {
   | --------------------- associated function in this implementation
...
28 |     pub fn new(num_elements: usize, exponent: f64) -> Result<Self, ()> {
   |            ^^^

[RUSTC-TIMING] alloctests test:true 8.748
error: could not compile `alloctests` (test "alloctests") due to 26 previous errors
error: constant `EMPTY_MAX` is never used
##[error]    --> coretests/tests/slice.rs:2489:7
     |
2489 | const EMPTY_MAX: &'static [()] = &[(); usize::MAX];
     |       ^^^^^^^^^
     |
     = note: `-D dead-code` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(dead_code)]`

[RUSTC-TIMING] coretests test:true 44.433
error: could not compile `coretests` (test "coretests") due to 6 previous errors
env -u RUSTC_WRAPPER CARGO_ENCODED_RUSTDOCFLAGS="-Csymbol-mangling-version=v0\u{1f}-Zrandomize-layout\u{1f}-Zunstable-options\u{1f}--check-cfg=cfg(bootstrap)\u{1f}--check-cfg=cfg(llvm_enzyme)\u{1f}-Dwarnings\u{1f}-Wrustdoc::invalid_codeblock_attributes\u{1f}--crate-version\u{1f}1.88.0-nightly\t(98e7afc0a\t2025-04-01)" CARGO_ENCODED_RUSTFLAGS="-Csymbol-mangling-version=v0\u{1f}-Zrandomize-layout\u{1f}-Zunstable-options\u{1f}--check-cfg=cfg(bootstrap)\u{1f}--check-cfg=cfg(llvm_enzyme)\u{1f}-Zmacro-backtrace\u{1f}-Csplit-debuginfo=off\u{1f}-Clink-arg=-L/usr/lib/llvm-18/lib\u{1f}-Cllvm-args=-import-instr-limit=10\u{1f}-Clink-args=-Wl,-z,origin\u{1f}-Clink-args=-Wl,-rpath,$ORIGIN/../lib\u{1f}-Alinker-messages" RUSTC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/dist/rustc-clif" RUSTDOC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/dist/rustdoc-clif" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--manifest-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/build/sysroot_tests/Cargo.toml" "--target-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/build/sysroot_tests_target" "--locked" "--target" "x86_64-unknown-linux-gnu" "-p" "coretests" "-p" "alloctests" "--tests" "--" "-q" exited with status ExitStatus(unix_wait_status(25856))
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:31:36
  local time: Tue Apr  1 14:32:07 UTC 2025
  network time: Tue, 01 Apr 2025 14:32:07 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

BoxyUwU commented:
r? compiler

BoxyUwU commented:
r? compiler

lcnr commented:
r? compiler

jieyouxu commented:
r? jieyouxu

bjorn3 commented:
> cg_clif's build system passes `--cap-lints=allow`, but somehow building coretests and alloctests is still emitting warnings.

Ah, only for the extended_sysroot test suite, but I moved coretests and alloctests to the base_sysroot test suite. Added an explicit `--cap-lints=allow` for them.

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14242546195/job/39915683633) [(plain)](https://github.com/rust-lang/rust/commit/13970949d135f3b0e1b9ece433c5fc230804d657/checks/39915683633/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#19 exporting to docker image format
#19 sending tarball 20.1s done
#19 DONE 26.9s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'build.print-step-timings', '--enable-verbose-tests', '--set', 'build.metrics', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: build.build          := x86_64-unknown-linux-gnu
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
  Number of decisions:   4447
  longest path:          1159 (code:    152)
  longest backtrack:       66 (code:    428)
Shared 86733 out of 152951 states by creating 14756 new states, saving 71977
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc: In function ‘rtx_def* extract_bit_field_1(rtx, poly_uint64, poly_uint64, int, rtx, machine_mode, machine_mode, bool, bool, rtx_def**)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc:1864:45: warning: ‘*(unsigned int*)((char*)&imode + offsetof(scalar_int_mode, scalar_int_mode::m_mode))’ may be used uninitialized [-Wmaybe-uninitialized]
 1864 |       rtx sub = extract_bit_field_as_subreg (mode1, op0, imode,
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
 1865 |                                              bitsize, bitnum);
      |                                              ~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/expmed.cc:1824:19: note: ‘*(unsigned int*)((char*)&imode + offsetof(scalar_int_mode, scalar_int_mode::m_mode))’ was declared here
 1824 |   scalar_int_mode imode;
      |                   ^~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gimple-range-gori.cc: In member function ‘void range_def_chain::dump(FILE*, basic_block, const char*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gimple-range-gori.cc:319:19: warning: format not a string literal and no format arguments [-Wformat-security]
  319 |           fprintf (f, prefix);
      |           ~~~~~~~~^~~~~~~~~~~
---
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc:7930:9: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 7930 |   write (fd, "\n\n", 2);
      |   ~~~~~~^~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc: In member function ‘void driver::final_actions() const’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/gcc.cc:9307:13: warning: ignoring return value of ‘int truncate(const char*, __off_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 9307 |     truncate(totruncate_file, 0);
      |     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto-wrapper.cc: In function ‘bool find_and_merge_options(int, off_t, const char*, vec<cl_decoded_option>, bool, vec<cl_decoded_option>*, const char*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto-wrapper.cc:1165:8: warning: ignoring return value of ‘ssize_t read(int, void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1165 |   read (fd, data, length);
      |   ~~~~~^~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc: In function ‘void lto_resolution_read(splay_tree, FILE*, lto_file*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2091:10: warning: ignoring return value of ‘int fscanf(FILE*, const char*, ...)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2091 |   fscanf (resolution, " ");   /* Read white space.  */
      |   ~~~~~~~^~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2093:9: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2093 |   fread (obj_name, sizeof (char), name_len, resolution);
      |   ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/lto/lto-common.cc:2113:10: warning: ignoring return value of ‘int fscanf(FILE*, const char*, ...)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 2113 |   fscanf (resolution, "%u", &num_symbols);
      |   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/gcc/jit/jit-recording.cc:32:
---
Applying io_quotes_use            to linux/blkzoned.h
Applying io_quotes_use            to linux/ipmi.h
Applying io_quotes_use            to linux/psp-dbc.h
Applying io_quotes_use            to linux/bt-bmc.h
Applying io_quotes_use            to linux/tps6594_pfsm.h
Applying io_quotes_use            to linux/cxl_mem.h
Applying io_quotes_use            to linux/wmi.h
Applying io_quotes_use            to linux/auto_fs.h
Applying io_quotes_use            to linux/mmtimer.h
Applying io_quotes_use            to linux/f2fs.h
Applying io_quotes_use            to linux/vhost.h
---
Applying machine_name             to x86_64-linux-gnu/bits/unistd_ext.h
Applying io_quotes_use            to x86_64-linux-gnu/asm/mtrr.h
Applying io_quotes_use            to x86_64-linux-gnu/asm/amd_hsmp.h
Applying machine_name             to openssl/e_os2.h
Applying io_quotes_use            to drm/xe_drm.h
Applying io_quotes_use            to drm/radeon_drm.h
Applying io_quotes_use            to drm/panfrost_drm.h
Applying io_quotes_use            to drm/etnaviv_drm.h
Applying io_quotes_use            to drm/lima_drm.h
Applying io_quotes_use            to drm/qaic_accel.h
Applying io_quotes_use            to drm/vc4_drm.h
Applying io_quotes_use            to drm/i915_drm.h
Applying io_quotes_use            to drm/omap_drm.h
Applying io_quotes_use            to drm/pvr_drm.h
Applying io_quotes_use            to drm/amdgpu_drm.h
Applying io_quotes_use            to drm/vgem_drm.h
Applying io_quotes_use            to drm/msm_drm.h
Applying io_quotes_use            to drm/v3d_drm.h
Applying io_quotes_use            to drm/exynos_drm.h
Applying io_quotes_use            to drm/nouveau_drm.h
Applying io_quotes_use            to drm/drm.h
Applying io_quotes_use            to drm/habanalabs_accel.h
Applying io_quotes_use            to drm/tegra_drm.h
Applying io_quotes_use            to rdma/rdma_user_ioctl.h
cc1: note: self-tests are not enabled in this build
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/c++tools/server.cc: In function ‘void server(bool, int, module_resolver*)’:
/checkout/obj/build/x86_64-unknown-linux-gnu/gcc/src/c++tools/server.cc:620:10: warning: ignoring return value of ‘int pipe(int*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
---
##[endgroup]
##[group][AOT] raw-dylib
[AOT] raw-dylib
##[endgroup]
##[group][TEST] sysroot
[TEST] sysroot
[COPY] sysroot_tests source
[PATCH] "sysroot_tests" <- "0027-sysroot_tests-128bit-atomic-operations.patch"
[PATCH] "sysroot_tests" <- "0028-sysroot_tests-Disable-long-running-tests.patch"
---
    collections::btree::node::tests::test_sizes

test result: FAILED. 230 passed; 1 failed; 16 ignored; 0 measured; 0 filtered out; finished in 2.90s

error: test failed, to rerun pass `-p alloctests --lib`
env -u RUSTC_WRAPPER CARGO_ENCODED_RUSTDOCFLAGS="-Csymbol-mangling-version=v0\u{1f}-Zrandomize-layout\u{1f}-Zunstable-options\u{1f}--check-cfg=cfg(bootstrap)\u{1f}--check-cfg=cfg(llvm_enzyme)\u{1f}-Dwarnings\u{1f}-Wrustdoc::invalid_codeblock_attributes\u{1f}--crate-version\u{1f}1.88.0-nightly\t(8090391e2\t2025-04-03)" CARGO_ENCODED_RUSTFLAGS="-Csymbol-mangling-version=v0\u{1f}-Zrandomize-layout\u{1f}-Zunstable-options\u{1f}--check-cfg=cfg(bootstrap)\u{1f}--check-cfg=cfg(llvm_enzyme)\u{1f}-Zmacro-backtrace\u{1f}-Csplit-debuginfo=off\u{1f}-Clink-arg=-L/usr/lib/llvm-18/lib\u{1f}-Cllvm-args=-import-instr-limit=10\u{1f}-Clink-args=-Wl,-z,origin\u{1f}-Clink-args=-Wl,-rpath,$ORIGIN/../lib\u{1f}-Alinker-messages\u{1f}--cap-lints=allow" RUSTC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/dist/rustc-clif" RUSTDOC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/dist/rustdoc-clif" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--manifest-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/build/sysroot_tests/Cargo.toml" "--target-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_clif/build/sysroot_tests_target" "--locked" "--target" "x86_64-unknown-linux-gnu" "-p" "coretests" "-p" "alloctests" "--tests" "--" "-q" exited with status ExitStatus(unix_wait_status(25856))
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:30:44
  local time: Thu Apr  3 12:41:36 UTC 2025
  network time: Thu, 03 Apr 2025 12:41:36 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

bjorn3 committed:
Fix testing with randomized layouts enabled

jieyouxu reviewed (approved):
Thanks

jieyouxu commented:
@bors r+ rollup=iffy

bors commented:
:pushpin: Commit fde54c2c0301be851543220ce64c74a468a17311 has been approved by `jieyouxu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=jieyouxu fde54c2c0301be851543220ce64c74a468a17311 -->
<!-- homu: {"type":"Approved","sha":"fde54c2c0301be851543220ce64c74a468a17311","approver":"jieyouxu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/14264835553/job/39995593973)
Approved by: jieyouxu
Pushing b8ff7b682e0544de40a90bb9cbd2d21f479b4987 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"jieyouxu","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/14264835553/job/39995593973"},"merge_sha":"b8ff7b682e0544de40a90bb9cbd2d21f479b4987"} -->

rust-timer commented:
Finished benchmarking commit (b8ff7b682e0544de40a90bb9cbd2d21f479b4987): [comparison URL](https://perf.rust-lang.org/compare.html?start=a4166dabaae56ab0d35a7825c3e7008778eacf34&end=b8ff7b682e0544de40a90bb9cbd2d21f479b4987&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=a4166dabaae56ab0d35a7825c3e7008778eacf34&end=b8ff7b682e0544de40a90bb9cbd2d21f479b4987&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=a4166dabaae56ab0d35a7825c3e7008778eacf34&end=b8ff7b682e0544de40a90bb9cbd2d21f479b4987&stat=max-rss)
<details>
<summary>Results (secondary -2.0%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -2.0% | [-2.3%, -1.6%] | 2     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=a4166dabaae56ab0d35a7825c3e7008778eacf34&end=b8ff7b682e0544de40a90bb9cbd2d21f479b4987&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=a4166dabaae56ab0d35a7825c3e7008778eacf34&end=b8ff7b682e0544de40a90bb9cbd2d21f479b4987&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 780.575s -> 781.07s (0.06%)
**Artifact size**: 365.99 MiB -> 365.97 MiB (-0.00%)
